kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/SectionEditor.vue

```diff
@@ -1,21 +1,405 @@
+          <KButton
+            :primary="true"
+            :text="applySettings$()"
+            class="apply-settings-style"
+            @click="quizForge.updateSection()"
+          />
+
+        </KGridItem>
+      </KGrid>
+    </div>
   </div>
 
 </template>
 
 
 <script>
 
+  import { enhancedQuizManagementStrings } from 'kolibri-common/strings/enhancedQuizManagementStrings';
+  import useKResponsiveWindow from 'kolibri.coreVue.composables.useKResponsiveWindow';
+  import Draggable from 'kolibri.coreVue.components.Draggable';
+  import DragContainer from 'kolibri.coreVue.components.DragContainer';
+  import DragHandle from 'kolibri.coreVue.components.DragHandle';
+
   export default {
     name: 'SectionEditor',
+    components: {
+      Draggable,
+      DragContainer,
+      DragHandle,
+    },
+    inject: ['quizForge'],
+    setup() {
+      const {
+        sectionSettings$,
+        sectionTitle$,
+        numberOfQuestionsLabel$,
+        optionalDescriptionLabel$,
+        quizResourceSelection$,
+        numberOfSelectedResources$,
+        currentSection$,
+        deleteSection$,
+        applySettings$,
+        changeResources$,
+        sectionOrder$,
+        questionOrder$,
+        randomizedLabel$,
+        randomizedOptionDescription$,
+        fixedLabel$,
+        fixedOptionDescription$,
+      } = enhancedQuizManagementStrings;
+      const { windowIsLarge, windowIsSmall } = useKResponsiveWindow();
+      return {
+        windowIsLarge,
+        windowIsSmall,
+        sectionSettings$,
+        sectionTitle$,
+        numberOfQuestionsLabel$,
+        optionalDescriptionLabel$,
+        quizResourceSelection$,
+        numberOfSelectedResources$,
+        currentSection$,
+        deleteSection$,
+        applySettings$,
+        changeResources$,
+        sectionOrder$,
+        questionOrder$,
+        randomizedLabel$,
+        randomizedOptionDescription$,
+        fixedLabel$,
+        fixedOptionDescription$,
+      };
+    },
+    data() {
+      return {
+        selectedQuestionOrder: true,
+        numQuestions: 1,
+        descriptionLabel: '',
+        sectionTitle: '',
+      };
+    },
+    computed: {
+      borderStyle() {
+        return `border: 1px solid ${this.$themeTokens.fineLine}`;
+      },
+      activeSection() {
+        return {
+          backgroundColor: this.$themePalette.grey.v_50,
+          border: `1px solid ${this.$themeTokens.fineLine}`,
+        };
+      },
+      dividerStyle() {
+        return `color : ${this.$themeTokens.fineLine}`;
+      },
+      sectionOrderList() {
+        return [
+          {
+            name: 'Section order',
+            current: this.currentSection$(),
+            isActive: true,
+          },
+          {
+            name: 'Section 2 / unique title',
+            current: null,
+            isActive: false,
+          },
+        ];
+      },
+    },
+    methods: {
+      handleSectionSort() {
+        this.quizForge.updateQuiz();
```

Comment:
You'll need to pass something to update here.

So, that `@sort` event that calls this `handleSectionSort` will pass a value (I think just the new version of the list values that have been sorted) -- you'll need to make sure that the list has objects that validate with the type defined as a `QuizSection`. So here you should basically expect to receive a parameter in this function.

The shape of the data given to this parameter, I believe, will be the same shape you define in `sectionOrderList` (which should be returning `quizForge.allSections`) -- so this means that if you're using that `allSections` list, which is a list of objects in [this shape](https://github.com/learningequality/kolibri/blob/16a9422e69a4f94782e3bcba3ba44539f84694b4/kolibri/plugins/coach/assets/src/composables/quizCreationSpecs.js#L115), then you can just take the value of that given parameter and pass that to the `updateQuiz` call.

Note how on the `Quiz` [typedef](https://github.com/learningequality/kolibri/blob/16a9422e69a4f94782e3bcba3ba44539f84694b4/kolibri/plugins/coach/assets/src/composables/quizCreationSpecs.js#L163-L166) you'll see there is a `question_sources` property which is where we store the quiz's list of `QuizSection` objects.

This means that the [`updateQuiz`](https://github.com/learningequality/kolibri/blob/16a9422e69a4f94782e3bcba3ba44539f84694b4/kolibri/plugins/coach/assets/src/composables/useQuizCreation.js#L177-L182) needs an object with keys that map to the Quiz type. In this particular case, you should be able to take in the param passed to `handleSectionSort` and then pass `{ question_sources: theParamThatWasPassedIn }` to update the order of the sections.